yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/call/ImageResourceFetcher.java

```diff
@@ -0,0 +1,55 @@
+package com.yoti.api.client.spi.remote.call;
+
+import static com.yoti.api.client.spi.remote.call.YotiConstants.CONTENT_TYPE;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.CONTENT_TYPE_JPEG;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.CONTENT_TYPE_PNG;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import com.yoti.api.client.Image;
+import com.yoti.api.client.spi.remote.JpegAttributeValue;
+import com.yoti.api.client.spi.remote.PngAttributeValue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class ImageResourceFetcher {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ImageResourceFetcher.class);
+
+    private final RawResourceFetcher rawResourceFetcher;
+
+    public static ImageResourceFetcher newInstance(RawResourceFetcher rawResourceFetcher) {
+        return new ImageResourceFetcher(rawResourceFetcher);
+    }
+
+    private ImageResourceFetcher(RawResourceFetcher rawResourceFetcher) {
+        this.rawResourceFetcher = rawResourceFetcher;
+    }
+
+    Image doRequest(SignedRequest signedRequest) throws IOException, ResourceException {
+        SignedRequestResponse signedRequestResponse = rawResourceFetcher.doRequest(signedRequest);
+        String contentType = getContentType(signedRequestResponse.getResponseHeaders());
+        byte[] responseBody = signedRequestResponse.getResponseBody();
+        switch (contentType) {
+            case CONTENT_TYPE_PNG:
+                return new PngAttributeValue(responseBody);
+            case CONTENT_TYPE_JPEG:
+                return new JpegAttributeValue(responseBody);
+            default:
+                LOG.error("Failed to convert to PNG or JPEG Attribute Value");
```

Comment:
Usually I'm not a fan of error messages like this, which could forget to be added to if we add further types (e.g. "unsupported image type" might be better). But having said that, I can't foresee us adding support for any images any time soon, so I think this is fine.